deploy-service/common/src/main/java/com/pinterest/deployservice/handler/CommonHandler.java

```diff
@@ -196,14 +196,20 @@ String generateMessage(String buildId, EnvironBean envBean, DeployState state, D
         }
     }
 
-    public void sendChatMessage(String from, String rooms, String message, String color) {
+    public void sendChatMessage(String from, String rooms, String message, String color, String recipients) {
         if (StringUtils.isEmpty(rooms)) {
             return;
         }
         List<String> chatrooms = Arrays.asList(rooms.split(","));
 
         for (String chatroom : chatrooms) {
             try {
+                if (!StringUtils.isEmpty(recipients)) {
+                    List<String> targets = Arrays.asList(recipients.split(","));
```

Comment:
Since we don't have front-end validation, the `recipients` can be `bob,   jose,     `. `split()` will keep the empty spaces. Can Slack API properly handle these extra spaces, e.g. `@   bob  `?

Reply:
Make sense. The same improvement can apply to watch recipients too.

Reply:
We can have another PR to improve on all applicable fields.